tests/run.py

```diff
@@ -2,41 +2,134 @@
 
-def test_trinkets(klass : str, path : str):
-    if not args.test_trinkets:
-        return
+def get_talent_name(spec: WowSpec.WowSpec, talents: str):
     try:
-        from simc_support.game_data.WowSpec import get_wow_spec_from_combined_simc_name
-        from simc_support.game_data.Trinket import get_trinkets_for_spec
-    except ImportError:
-        print("Error importing simc-support library for trinket test. Skipping test.")
-        return
-
-    trinkets = get_trinkets_for_spec(
-        get_wow_spec_from_combined_simc_name(klass))
+        talent_row = next(i for i, x in enumerate(talents) if x != "0")
+        talent_choice = int(talents[talent_row])
+        talent = spec.talents[talent_row+1][talent_choice]
+        return talent["name"]
+    except StopIteration:
+        return "None"
+    except Exception:
+        return "Unknown"
+
+
+def test_talents(klass: str, path: str):
+    spec = WowSpec.get_wow_spec_from_combined_simc_name(klass)
+    for fight_style in FIGHT_STYLES:
+        grp = TestGroup('{}/{}/talents'.format(profile, fight_style),
+                        fight_style=fight_style, profile=path)
+        tests.append(grp)
+        for talents in talent_combinations(klass):
+            talent_name = get_talent_name(spec, talents)
+            Test('{:<40} ({})'.format(talent_name, talents),
+                 group=grp, args=[('talents', talents)])
+
+
+def test_covenants(klass: str, path: str):
+    spec = WowSpec.get_wow_spec_from_combined_simc_name(klass)
+    conduits = Conduit.get_conduits_for_spec(spec)
+    for fight_style in FIGHT_STYLES:
+        grp = TestGroup('{}/{}/covenants'.format(profile, fight_style),
+                        fight_style=fight_style, profile=path)
+        tests.append(grp)
+
+        # Test covenants
+        for covenant in Covenant.COVENANTS:
+            Test(covenant.full_name, group=grp,
+                 args=[('covenant', covenant.simc_name), ('level', 60)])
+            # Add conduits specific to selected covenant
+            for conduit in conduits:
+                if len(conduit.covenants) == 1 and covenant == conduit.covenants[0]:
+                    rank = random.choice(conduit.ranks) + 1
+                    soulbind_argument = '{}:{}'.format(conduit.id, rank)
+                    Test('{} - {} ({}) Rank {}'.format(covenant.full_name, conduit.full_name, conduit.id, rank), group=grp, args=[
+                        ('covenant', covenant.simc_name), ('level', 60), ('soulbind', soulbind_argument)])
+
+        # test conduits available for all 4 covenants independent from covenant selection.
+        for conduit in conduits:
+            if len(conduit.covenants) == 4:
+                rank = random.choice(conduit.ranks) + 1
+                soulbind_argument = '{}:{}'.format(conduit.id, rank)
+                Test('{} ({}) Rank {}'.format(conduit.full_name, conduit.id, rank), group=grp, args=[
+                    ('level', 60), ('soulbind', soulbind_argument)])
```

Comment:
One thought occurred to me while thinking about the profiles: this won't actually enable the conduits because there is no covenant set afaik. Which is a bummer...
Could just bite the bullet and just test all potency conduits with all covenants. Will add 12 more tests, but at least we'll know they def work.